main/src/observables/time_energy_growth.hpp

```diff
@@ -30,78 +30,74 @@
  * @author Lukas Schmidt
  */
 
+#include <array>
+#include <mpi.h>
+
 #include "iobservables.hpp"
 #include "sph/math.hpp"
-#include <mpi.h>
 #include "io/ifile_writer.hpp"
 
+namespace sphexa
+{
 
-namespace sphexa {
+template<class Tc, class Tm>
+std::array<Tc, 3> localGrowthRate(size_t startIndex, size_t endIndex, const Tc* x, const Tc* y, const Tm* vy,
+                                  const Tm* rho, const Tm* m, const Tm* kx, const cstone::Box<Tc>& box)
+{
+    const Tc ybox = box.ly();
 
+    Tc sumsi = 0;
+    Tc sumci = 0;
+    Tc sumdi = 0;
 
-template<typename T, class Dataset>
-void localGrowthRate(size_t startIndex, size_t endIndex, Dataset& d, T* sumsi, T* sumci, T* sumdi, const cstone::Box<T>& box)
-{
```

Comment:
This function only needs arrays for `x, y, vy, rho, m, kx`, so we pass that directly instead of the whole Dataset,
achieving better code modularity. If we want to write a test for this, we now only have to supply the needed arrays,
rather than create an entire Dataset.